File: app/src/main/java/com/test/xyz/demo/presentation/projectdetails/di/RepoDetailsFragmentModule.java
package com.test.xyz.demo.presentation.projectdetails.di;

import com.test.xyz.demo.presentation.common.di.scope.ActivityScope;
import com.test.xyz.demo.presentation.projectdetails.presenter.ProjectDetailsView;
import com.test.xyz.demo.presentation.projectdetails.presenter.ProjectDetailsPresenter;
import com.test.xyz.demo.presentation.projectdetails.presenter.ProjectDetailsPresenterImpl;

import dagger.Module;
import dagger.Provides;

@Module
public class RepoDetailsFragmentModule {
    private final ProjectDetailsView view;

    public RepoDetailsFragmentModule(ProjectDetailsView view) {
        this.view = view;
    }

    @Provides
    @ActivityScope
    ProjectDetailsView provideRepoDetailsView() {
        return this.view;
    }

    @Provides
    @ActivityScope
    ProjectDetailsPresenter provideRepoDetailsPresenter(ProjectDetailsPresenterImpl presenter) {
        return presenter;
    }
}

File: app/src/main/kotlin/com/test/xyz/demo/presentation/projectdetails/di/RepoDetailsFragmentModule.kt
package com.test.xyz.demo.presentation.projectdetails.di

import com.test.xyz.demo.presentation.common.di.scope.ActivityScope
import com.test.xyz.demo.presentation.projectdetails.presenter.ProjectDetailsPresenter
import com.test.xyz.demo.presentation.projectdetails.presenter.ProjectDetailsPresenterImpl
import com.test.xyz.demo.presentation.projectdetails.presenter.ProjectDetailsView
import dagger.Module
import dagger.Provides

@Module
class RepoDetailsFragmentModule(private val view: ProjectDetailsView) {

    @Provides
    @ActivityScope
    internal fun provideRepoDetailsView(): ProjectDetailsView {
        return this.view
    }

    @Provides
    @ActivityScope
    internal fun provideRepoDetailsPresenter(presenter: ProjectDetailsPresenterImpl): ProjectDetailsPresenter {
        return presenter
    }
}
